src/main/java/capstone/mfslbackend/service/TransactionService.java

```diff
@@ -237,10 +267,88 @@ public Boolean approveTeam(Set<Player> players) {
                 default -> throw new Error400("Invalid position for player with id " + player.getPlayerId());
             }
         }
-        return gkCount > MIN_GK && defCount > MIN_DEF && midCount > MIN_MID && fwdCount > MIN_FWD;
+        return gkCount > MIN_GK && defCount > DRAFT_MIN_DEF && midCount > DRAFT_MIN_MID && fwdCount > DRAFT_MIN_FWD;
     }
     public Transaction getTransactionById(Long id) {
         return transactionRepository.findById(id)
                 .orElseThrow(() -> new Error404("Transaction with id " + id + " not found"));
     }
+
+    public boolean validTransaction(FantasyTeam proposingFantasyTeam, Player outgoingPlayer, Player ingoingPlayer) throws Error404 {
+        String position = ingoingPlayer.getPosition();
+        Long fantasyTeamId = proposingFantasyTeam.getId();
+
+        String[] playerIdsArray = proposingFantasyTeam.getPlayerIdsInOrder().split(" ");
+        List<Long> playerIdsList = Arrays.stream(playerIdsArray)
+                .map(Long::parseLong)
+                .toList();
+
+        long totalDefenders = proposingFantasyTeam.getPlayers().stream()
+                .filter(player -> player.getPosition().equals("Defender"))
+                .count();
+
+        long startingDefenders = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Defender"))
+                .count();
+
+        long totalMidfielders = proposingFantasyTeam.getPlayers().stream()
+                .filter(player -> player.getPosition().equals("Midfielder"))
+                .count();
+
+        long startingMidfielders = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Midfielder"))
+                .count();
+
+        long totalAttackers = proposingFantasyTeam.getPlayers().stream()
+                .filter(player -> player.getPosition().equals("Attacker"))
+                .count();
+
+        long startingAttackers = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Attacker"))
+                .count();
+
+
+        long startingGoalkeepers = playerIdsList.stream()
+                .limit(PLAYERS_IN_STARTING_LINEUP)
+                .map(playerService::getPlayerById)
+                .filter(player -> player != null && player.getPosition().equals("Goalkeeper"))
+                .count();
+
+        if (position.equals("Goalkeeper") && startingGoalkeepers >= MIN_GK) {
```

Comment:
If you replace a gk with a gk this will trigger because you dont remove the outgoing player from these ids in startingGoalkeepers